addOns/automation/src/main/java/org/zaproxy/addon/automation/tests/UrlPresenceTest.java

```diff
@@ -0,0 +1,452 @@
+                        && isNullOrEmpty(responseHeaderRegex)) {
+                    return true;
+                }
+
+                requestheaderfound =
+                        checkPattern(
+                                requestHeaderRegex,
+                                msg.getRequestHeader().toString(),
+                                progress,
+                                operator);
+                requestbodyfound =
+                        checkPattern(
+                                requestBodyRegex,
+                                msg.getRequestBody().toString(),
+                                progress,
+                                operator);
+                responseheaderfound =
+                        checkPattern(
+                                responseHeaderRegex,
+                                msg.getResponseHeader().toString(),
+                                progress,
+                                operator);
+                responsebodyfound =
+                        checkPattern(
+                                responseBodyRegex,
+                                msg.getResponseBody().toString(),
+                                progress,
+                                operator);
+
+                if (operator.equalsIgnoreCase("and")) {
+                    result =
+                            requestheaderfound
+                                    && requestbodyfound
+                                    && responseheaderfound
+                                    && responsebodyfound;
+                } else if (operator.equalsIgnoreCase("or") || isNullOrEmpty(operator)) {
+                    result =
+                            requestheaderfound
+                                    || requestbodyfound
+                                    || responseheaderfound
+                                    || responsebodyfound;
+                } else {
+                    String badoperatormessage =
+                            Constant.messages.getString(
+                                    "automation.tests.url.badOperator",
+                                    getJobType(),
+                                    getName(),
+                                    getTestType(),
+                                    operator);
+                    switch (onFail) {
+                        case INFO:
+                            progress.info(badoperatormessage);
+                            break;
+                        case WARN:
+                            progress.warn(badoperatormessage);
+                            break;
+                        case ERROR:
+                            progress.error(badoperatormessage);
+                            break;
+                        default:
+                            throw new RuntimeException("Unknown OnFail: " + onFail);
+                    }
+                }
+                return result;
```

Comment:
If a msg if found but doesnt match then that should be the same as not finding a message, right?
But in this case it looks like nothing is logged so the test wont actually fail...

Reply:
You can just return here if the test passes and fall through if it fails, and move the failing code out of the `else` statement...